pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -1594,31 +1533,64 @@ mod tests {
                 &LayerFileMetadata::new(content_3.len() as u64),
             )
             .unwrap();
+
+        {
+            let mut guard = client.upload_queue.lock().unwrap();
+            let upload_queue = guard.initialized_mut().unwrap();
+            assert_eq!(upload_queue.queued_operations.len(), 0);
+            assert_eq!(upload_queue.num_inprogress_layer_uploads, 1);
+        }
+
+        assert_remote_files(
+            &[
+                &layer_file_name_1.file_name(),
+                &layer_file_name_2.file_name(),
+                "index_part.json",
+            ],
+            &remote_timeline_dir,
+        );
+
         client
-            .schedule_layer_file_deletion(&[layer_file_name_1.clone()])
+            .schedule_layer_file_deletion(
+                &[layer_file_name_1.clone()],
+                &deletion_queue.new_client(),
+            )
+            .await
             .unwrap();
+
         {
             let mut guard = client.upload_queue.lock().unwrap();
             let upload_queue = guard.initialized_mut().unwrap();
 
-            // Deletion schedules upload of the index file, and the file deletion itself
-            assert!(upload_queue.queued_operations.len() == 2);
-            assert!(upload_queue.inprogress_tasks.len() == 1);
-            assert!(upload_queue.num_inprogress_layer_uploads == 1);
-            assert!(upload_queue.num_inprogress_deletions == 0);
-            assert!(upload_queue.latest_files_changes_since_metadata_upload_scheduled == 0);
+            // Deletion schedules upload of the index file via RemoteTimelineClient, and
+            // deletion of layer files via DeletionQueue.  The uploads have all been flushed
+            // because schedule_layer_file_deletion does a wait_completion before pushing
+            // to the deletion_queue
+            assert_eq!(upload_queue.queued_operations.len(), 0);
+            assert_eq!(upload_queue.inprogress_tasks.len(), 0);
+            assert_eq!(upload_queue.num_inprogress_layer_uploads, 0);
+            assert_eq!(upload_queue.num_inprogress_deletions, 0);
```

Comment:
1. I can't find any place where `num_inprogress_deletions` is or would become != 0. Let's remove it?
2. Is the upload queue delete op even still relevant? ISTM that we replaced the upload queue delete op and associated rules in the remote_timeline_client with wait_completion calls at the place where we schedule deletes. Right?

Reply:
> can't find any place where num_inprogress_deletions is or would become != 0. Let's remove it?

Yep, it's vestigial.  Removed in 1afc6337fb8220d33a6fc640c8704841f22bfb86

This also reminds me that the RemoteTimelineClient has some comments about deletion that needed updating d9755becab9608afc6685c17801751c5b98b405a

>Is the upload queue delete op even still relevant? ISTM that we replaced the upload queue delete op and associated rules in the remote_timeline_client with wait_completion calls at the place where we schedule deletes. Right?

Right.  It's removed.